Binyamina manual deployment guide (#50)
RegattaData/docs: self-hosted-deployment/manual-deployment-copied-1/configuration-file.mdx

---

> title: "Deployment Setup"

Reviewer: File name should match the page title.

Author: It was - THEN mintlify lost my entire file. Hack no way.
Once evrything in and committed I'll make these changes

---

## Configuration Planning

> First, we need to do is define the topology of the system. This topology is derived from the workload profile, performance goals and storage requirements from the RegattaDB. Since RegattaDB is a versatile system, we can configure it based on the different needs.

Reviewer: Rephrased:

The first step is defining the system topology, based on the workload profile, performance targets, and storage requirements. RegattaDB’s flexible architecture allows the system to be tailored to different deployment scenarios.

> Below are several common configurations.

Reviewer: Before common configuration we need a section with "Placement rules" where we give a systematic overview of the module placement rules. 
In addition, in this section I would start by formally defining the term "node".
Then bullets with our placement rules:
- Single SM module deployed per cluster. Can be deployed alongside other modules or alone.
- Every operational module must by accompanied by SNA
- GDD, SEQUENCER, GDD operational modules - 1 per cluster
- RDB - 1 or more 
- etc.
I would also add a statement saying that when planning a multi node deployment, we might recommend different module placement rules depending on the performance objectives, and different hardware recommendations (RAM, CPU, Storage, Network) accordingly. Our performance experts can assist with coming up with the best deployment plan tailored to your needs. We might also want to refer to Tomer Florentz's page re best practices here..

#### Single Node

<Frame>
  ![Image](/images/image-1.png)

> </Frame>

Reviewer: I would remove the internal and external IP + bus illustrations and internal network stuff, they are not used in the way we define these internally are don't add to the point we are trying to make. It is worth talking about this topic in general (networking) somewhere else but I am not sure it should be in the deployment guide scope at all (or at least not for now.. )

Other comments on the illustrations (I leave it here but applies to all):
- Change "Server" to "Node" as this is the terminology we use throughout . 
- The title "Server" (which will be changed to "Node") should be in a visible color. In the single node example it looks ok, but in the rest your barely see the title since its in gray with gray background.
- Remove the white background/border around the illustration so it blends seamlessly with the page.
- Add a short text below each example.

> #### Multi-Node – Common Performance

Reviewer: I wouldn't use the word "Performance" . 
Maybe just "Multi-Node" 
And then text stating this is the "classic" scale out configuration.

<Frame>
  ![Image](/images/image-2.png)
</Frame>

> #### Multi-node - Distributed

Reviewer: Again, I would remove "Distributed" as this word has a double meaning. All of the multi node configurations are "distributed" as we use this term in different ways. I would emphasize that this configuration may be optimal for performance sensitive use cases, where we would like to dedicate the most resources for Regatta modules. This can become more relevant as the cluster scales. I would also add more RDB modules to this example (more than the previous one at least). Also, you have GDD twice.

In this case, we distribute the non-RDB modules across different nodes.

<Frame>
  ![Image](/images/image-3.png)
</Frame>

> Note that in this case we have assigned a dedicated server for the SM. This is a common configuration in a Kubernatis system \<\<\<link\>\>\>. Since there’s only an SM process on the that server, we don’t need an SNA. Also, since we did not deploy an RDB, the resource requirements from this server are minimal.

Reviewer: This should not be the opening sentence, but rather the last one in the paragraph after the main explanation about this configuration (my previous comment). 
Also I would phrase it differently:
"Note that since we place the SM module alone on this node, we don’t need an SNA. This topology may be common in some use cases, such as Kubernetes (K8) deployments". 
Remove the last sentence re minimal requirements.

For guidelines on sizing and configuring the RegattaDB to cater to your needs, please refer to \<\<\<BEST PRACTICES\>\>\> or contact support for assistance.

### Hardware Requirements

> You will now need server to accommodate you RegattaDB configuration. These servers may be physical, VMs or even containers. They need to at least meet all the prerequisites defined previously:

Reviewer: The phrasing here is off, also again usage of server instead of node. You actually implicitly define here what I requested that will be defined above which is "what is a node" . 
So I would move this part above, and then the rest is moot (talking about the RAM, CPU etc.). The last sentence of the previous section is:
 For guidelines on sizing and configuring the RegattaDB to cater to your needs, please refer to \<\<\<BEST PRACTICES\>\>\> or contact support for assistance.
And then you have this section where you write again:
To understand the exact requirements, you will need to size the system based on your workload needs. You can consult our \<\<\<BEST PRACTICES\>\>\> documentation or contact Regatta support.

So in total I would remove this whole section and add the node definition above. Then leave the final sentence that can be a merge the 2 almost identical sentences re making sure nodes are configured for RegattaDB installation per our prerequisites , and refer to best practices or support for custom configurations.

Author: Tried fixing most of it. Hopefully covered

- CPUs
- RAM
- Storage devices Networking

To understand the exact requirements, you will need to size the system based on your workload needs. You can consult our \<\<\<BEST PRACTICES\>\>\> documentation or contact Regatta support.

### Regatta Software

> Next, you will need the RegattaDB software that you received from our support team. That package contains the software for all RegattaDB modules described above plus other useful tools.

Reviewer: Rephrase suggestion:
Next, download the RegattaDB software package. RegattaDB is distributed as a set of RPM packages, each corresponding to a specific module or tool. These RPMs should be copied to the deployment nodes, where they will be installed in later steps.

The package includes all core modules required to run the system, along with additional tools and client components.

Example package contents:

Below is an example of the content of the package file.

#### Deployment Software

```shellscript
regatta-dcm-26.0.0.112-1.el8.x86_64.rpm
regatta-gdd-26.0.0.112-1.el8.x86_64.rpm
regatta-rdb-26.0.0.112-1.el8.x86_64.rpm
regatta-sequencer-26.0.0.112-1.el8.x86_64.rpm
regatta-sm-26.0.0.112-1.el8.x86_64.rpm
regatta-sna-26.0.0.112-1.el8.x86_64.rpm
```

#### Tools

```shellscript
regatta-bnr-26.0.0.112-1.el8.x86_64.rpm
regatta-rdt-26.0.0.112-1.el8.x86_64.rpm

> ```

Reviewer: Please add a short explanation about each package here.
bnr - The Regatta Backup & Restore Tool
rdt - The Regatta deployment tool

Author: Done

#### Clients

```shellscript
regatta-cli-26.0.0.112-1.el8.x86_64.rpm
regatta-client-26.0.0.112-1.el8.x86_64.rpm
regatta-connect-26.0.0.112-1.el8.x86_64.rpm

> ```

Reviewer: Please add a short explanation about each package here.
cli - The Regatta CLI Tool
client - The Regatta client library
connect – Full suite of drivers and connectors for integrating RegattaDB with external systems and frameworks.

Author: Done

#### Debugging Tools

```shellscript
regatta-debug-26.0.0.112-1.el8.x86_64.rpm
```

> For details see \<\<\<links\>\>\>

Reviewer: Where are you referring to?

### EULA Acceptance

> To deploy and use RegattaDB, you must read and accept the **Regatta End User License Agreement (EULA)**. This acceptance is built directly into the RPM installation process; you will be prompted to agree to the terms before the installation can be completed.

Reviewer: Rephrase suggestion:
To deploy and use RegattaDB, you must accept the Regatta End User License Agreement (EULA). Acceptance is enforced as part of the RPM installation process.

> - **Interactive Installs**: yum install will prompt for EULA acceptance.

Reviewer: ```suggestion
- **Interactive Installs** (e.g., yum install, dnf install): you will be prompted to accept the EULA during installation.
```

> - **Automated Installs**: You must include the environment variable ACCEPT\_EULA=1.

Reviewer: ```suggestion
- **Automated Installs**: You must set the environment variable ACCEPT\_EULA=1 to proceed without prompts.
```

Author: Done

> Please take a few minutes to read and approve the Regatta End User License Agreement at the Regatta website ([https://regatta.dev/eula/](https://regatta.dev/eula/)) before deployment, especially if you are using the environment variable to bypass interactive prompts.

Reviewer: Rephrase suggestion:
Before deployment, review the EULA at: https://regatta.dev/eula/

This is especially important when using automated installations where prompts are bypassed.

> 

Reviewer: There are 2 missing sections (by order)
1. Storage permissions - You can refer to old deployment guide and copy I think.
2. Installing RPMs - Installing the RPMs, and the resulting structure under /opt/regatta (show with linux command tree -L). In this section we should also talk about the SM and SNA being actual linux services that are responsible for the deployment of the rest of the cluster modules. Also need to show the START_SERVICE=1 option that starts the services right away upon installation . 
3. Firewall stuff - can be probably copied from old docs.

Author: They do appear - the details are in the example.
Less redundancy

### Deployment User

> As with any software deployment, we require a user which has all the required permissions, deployment and storage access.

Reviewer: I would remove this sentence

In the installation process, the RPM creates a user named **regatta** with required permissions and establishes the directory structure. If you prefer to use another user, say **my-user**, please perform the following steps:

- Make sure **my-user** has the required permissions
- Transfer ownership of the created directory structure
- Add **my-user** to the required permissions groups (if missing)

> - Update the System Configuration files for both SM and SNA

Reviewer: These are not at all the steps we talked about that are needed to work with a custom user that is not the default regatta user. 
You should be very specific here about the steps:
- Changing /opt/regatta permissions
- Changing /var/log/regatta permissions
- Changing storage permissions
- Changing the user and group in the relevant service files

---

> title: "Deployment Setup"

Reviewer: Need to update the file name

---

> ## Configuration Planning

Reviewer: ```suggestion
## Deployment Planning
```

> The first step is defining the system's topology, which is determined by your workload profile, performance goals, and storage requirements. Since RegattaDB is highly versatile, we can tailor this configuration to meet specific needs.

Reviewer: ```suggestion
The first step is defining your deployment topology. The optimal topology depends on your workload profile, performance objectives, and storage requirements. RegattaDB supports a wide range of deployment configurations, allowing the system to be tailored to your specific requirements.
```

> The basic guidelines are:

Reviewer: ```suggestion
General deployment guidelines:
```

- Management Modules

>   - SM - There is only a single SM process in RegattaDB

Reviewer: ```suggestion
  - SM - There is only a single SM process in every RegattaDB cluster
```

>   - SNA - Should be deployed on any node where's at least one Operational Module

Reviewer: ```suggestion
  - SNA - Deploy on every node that hosts at least one Operational Module
```

- Operational Modules

>   - RDB - One on node server that will manage DB operations

Reviewer: This is not correct. We can have more than 1 rdb per node. I would simply write "at least one RDB module"

>   - GDD, DCM, Sequencer - one of each

Reviewer: ```suggestion
  - GDD, DCM, Sequencer - exactly one of each
```

Below are several common configurations.

> #### Single Node

Reviewer: Dd, this is a general comment for all the cluster illustrations - they are not good. 
The graphics are bad, is contains the port stuff which is both wrong and redundant. Let's find a way to improve the graphics. Please ask @topazpalgi for help.

> In a single node deployment, there is only one option:

Reviewer: ```suggestion
A single-node deployment consists of a single server hosting all RegattaDB modules:
```

<Frame>
  ![Image](/images/image-1.png)
</Frame>

#### Multi-Node – Classic

The most common multi-node deployment is:

- One node holds all modules as a Single Node deployment will do.

> - All other nodes have the RDB and the SNA

Reviewer: The most common multi-node deployment uses the following topology:

One node hosts all RegattaDB modules, as in a single-node deployment.
All additional nodes host both the RDB and SNA modules.

<Frame>
  ![Image](/images/image-2.png)
</Frame>

> In this case,  RegattaDB gets the resources available and utilizes all the nodes.

Reviewer: Remove

> #### Multi-node - Standalone

Reviewer: Please remove this section, leave only the last sentence 
"For guidelines on sizing and configuring the RegattaDB to cater to your needs, please refer to \<\<\<BEST PRACTICES\>\>\> or contact support for assistance."

In this case, we dedicate a node to the SM and also distribute the non-RDB Operational Modules across different nodes.

<Frame>
  ![Image](/images/image-5.png)
</Frame>

Note that since we place the SM module alone on this node, we don’t need an SNA. This topology may be common in some use cases, such as Kubernetes (K8) deployments.

For guidelines on sizing and configuring the RegattaDB to cater to your needs, please refer to \<\<\<BEST PRACTICES\>\>\> or contact support for assistance.

> ### Hardware Requirements

Reviewer: ```suggestion
### Provision the Deployment Nodes
```

My suggestion to rephrase this text:

Next, provision the nodes that will host your RegattaDB deployment. These may be physical servers, virtual machines (VMs), or containers, provided they meet the prerequisites [Add link] described in the previous section

The exact hardware requirements depend on your deployment topology and workload characteristics. For guidance on sizing your deployment, refer to Best Practices or contact Regatta Support for assistance.

You will now need node to accommodate you RegattaDB configuration. These nodes may be physical, VMs or even containers. They need to at least meet all the prerequisites defined previously:

- CPUs
- RAM

> - Storage devices Networking

Reviewer: Remove, put a link to the prerequisites
